Feature/vue update

nirazul opened the pull request:
@GianlucaGuarini 
Somehow the tests fail, but when I look at them in detail mode with `npm run debug` they are ok. Do you have any idea how to fix this?

This is a backwards incompatible change. So a major version bump is in order.

### Changelog:

- Replace `v-slot` syntax for `vue@2.6.x` support
- Tweak `countVNodes` method
- Misc text fixes
- Fix a scroll-lock-helper bug
- Remove modal component
- Simplify actions component to not try to strip text nodes (as there is no reliable way to do so)
- Update npm dependencies

GianlucaGuarini commented on package.json:
```diff
@@ -33,19 +33,19 @@
-    "@dreipol/eslint-config-vue": "^1.0.0",
-    "@vue/test-utils": "^1.0.0-beta.28",
+    "@dreipol/eslint-config-vue": "^2.0.2",
+    "@vue/test-utils": "^1.0.0-beta.29",
```
downgrade to `1.0.0-beta.28` and the tests should work again. I think they have broken something

nirazul replied:
Awesome hint! That saved me a lot of time.

GianlucaGuarini commented on src/components/form/input/input.vue:
```diff
@@ -1,16 +1,16 @@
     <div class="form-field form-field__input" :class="rootClasses">
         <label>
-            <span class="form-field--title-label" v-if="!hasFloatingLabel && $slots.label">
+            <span class="form-field--title-label" v-if="!hasFloatingLabel && $scopedSlots.label">
```
Are you sure that `$scopedSlots` will work also on old vue@2 versions?

nirazul replied:
I'm sure that it will break. But that's sort of the point. We'll enforce `vue@^2.6.0` with the major version bump. I've added a peer dependency that should throw a warning.

nirazul commented:
@GianlucaGuarini 
Can you check again? I've reverted to vue-test-utils and improved the coverage a bit. 
I also added tests for the `actions` component and updated its code, as the filtering of text-only tags is not feasible anymore. Vnodes can be anything and it's very hard to check if there's only text inside or not (for example, there could be a `template` node with a text child. So I've just removed the code for it as it seemed not that important for the functionality of the form field modules.

GianlucaGuarini commented:
@nirazul well done!
